Block/Catalog/Product/View.php

```diff
@@ -92,9 +92,11 @@ public function __construct(
      */
     private function getProduct()
     {
-        $this->product = $this->registry->registry('product');
-        if (!$this->product->getId()) {
-            throw new LocalizedException(__('Failed to initialize product'));
+        if(is_null($this->product)){
+            $this->product = $this->dataHelper->getProduct();
```

Comment:
Du coup, tu n'en profites pas pour tester si `$this->product` est `null` après l'avoir récupéré du `dataHelper` ?